unittests/test_pipeline.py

```diff
@@ -242,6 +242,27 @@ def test_supports_system(self):
         self.assertFalse(test.supports_system('testsys:gpu'))
         self.assertFalse(test.supports_system('testsys:login'))
 
+    def test_supports_environ(self):
+        test = self.loader.load_from_file(
+            'unittests/resources/hellocheck.py',
+            system=self.system, resources=self.resources
+        )[0]
+
+        test.valid_prog_environs = ['*']
+        self.assertTrue(test.supports_environ('foo1'))
+        self.assertTrue(test.supports_environ('foo-env'))
+        self.assertTrue(test.supports_environ('*'))
+
+        test.valid_prog_environs = ['PrgEnv-foo-*']
+        self.assertTrue(test.supports_environ('PrgEnv-foo-version1'))
+        self.assertTrue(test.supports_environ('PrgEnv-foo-version2'))
+        self.assertFalse(test.supports_environ('PrgEnv-boo-version1'))
+
+        test.valid_prog_environs = ['PrgEnv-*']
+        self.assertTrue(test.supports_environ('PrgEnv-foo1'))
+        self.assertTrue(test.supports_environ('PrgEnv-foo2'))
+        self.assertFalse(test.supports_environ('Prgenv-foo'))
```

Comment:
I don't see the reason for this set of checks. They are equivalent to the above block. You can simply add a check for case sensitivity.